RocketChat/Rocket.Chat.ReactNative
# [FIX] Unarchive permission not honored

## Proposed changes

Implemented to show an alert error when archive/unarchives a room.
Disable the `ARCHIVE/UNARCHIVE` button due to the role permission

## Issue(s)	

https://github.com/RocketChat/Rocket.Chat.ReactNative/issues/3118
I archive the channel, the "Unarchive" button appears, but it does not work
## How to test or reproduce


- Go to a Room (channel, dm, discussion)
- Click on Title
- At Actions screen, click on title
- And needs to appear the Edit button, then click on it
- Go to `Room Info Edit`
- Archive the room
- Unarchive the room
- If the button for Archive/Unarchive is blocked, it's because the user doesn't have permission to do that thing

## Screenshots

<table>
<thead>
<tr>
<th>Before</th>
<th>After</th>
</thead>
</tr>
<body>
<tr>
<td>
<img src="https://user-images.githubusercontent.com/47038980/123166236-90afad00-d44b-11eb-804c-caa9802dd587.png" height="450" />
</td>
<td>
<img src="https://user-images.githubusercontent.com/47038980/123166458-d0769480-d44b-11eb-9653-15c7d894771c.png"  height="450"/>
</td>
</tr>
<tr>
<td>
<img src="https://user-images.githubusercontent.com/47038980/123166324-a91fc780-d44b-11eb-9d18-17ed9d1ea15c.png" height="450" />
</td>
<td>
<img src="https://user-images.githubusercontent.com/47038980/123166328-aae98b00-d44b-11eb-9f0b-a6b79a395144.png"  height="450"/>
</td>
</tr>
</tbody>
</table>

## Types of changes




## Checklist



## Further comments

Closes https://github.com/RocketChat/Rocket.Chat.ReactNative/issues/3118
